lightning-background-processor/src/lib.rs

```diff
@@ -596,12 +638,33 @@ mod tests {
 	}
 
 	#[test]
-	fn test_persist_error() {
+	fn test_channel_manager_persist_error() {
 		// Test that if we encounter an error during manager persistence, the thread panics.
 		let nodes = create_nodes(2, "test_persist_error".to_string());
 		open_channel!(nodes[0], nodes[1], 100000);
 
-		let persister = |_: &_| Err(std::io::Error::new(std::io::ErrorKind::Other, "test"));
+		struct ChannelManagerErrorPersister {
+			data_dir: String,
+		}
+
+		impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L:Deref> super::Persister<Signer, M, T, K, F, L> for ChannelManagerErrorPersister where
+			M::Target: 'static + chain::Watch<Signer>,
+			T::Target: 'static + BroadcasterInterface,
+			K::Target: 'static + KeysInterface<Signer = Signer>,
+			F::Target: 'static + FeeEstimator,
+			L::Target: 'static + Logger,
+		{
+			fn persist_manager(&self, _channel_manager: &ChannelManager<Signer, M, T, K, F, L>) -> Result<(), std::io::Error> {
+				Err(std::io::Error::new(std::io::ErrorKind::Other, "test"))
```

Comment:
In tests, instead of erroring, we should panic to ensure the test fails.

Reply:
hm, this test is actually for testing if `channel_manager` persistence fails. I didn't change the [original implementation](https://github.com/lightningdevkit/rust-lightning/pull/1376/files#diff-3c70a8bcbb58522dbe589e774a85bbb611adab7142bb1eacec4fb7828cb3ee06L604) of `persist_manager` in tests. 

Right now it [checks and panics if there isn't an error](https://github.com/lightningdevkit/rust-lightning/pull/1376/files#diff-3c70a8bcbb58522dbe589e774a85bbb611adab7142bb1eacec4fb7828cb3ee06R670-R675).

Reply:
actually that makes me wonder, and I apologize if this is a dumb question: since we only do `log_error!` instead of throwing it when persisting a graph fails, how do we test that behavior? do I use the `testing_logger` crate?

Reply:
Oops lol sorry.

> how do we test that behavior?

Our `TestLogger` struct has a few `assert_log*` functions that allow you to assert a regex or specific string was logged. We don't generally worry *too* much about ensuring particular log entries are printed, though.

Reply:
@TheBlueMatt gotcha -- in that case, I created [a test](https://github.com/lightningdevkit/rust-lightning/pull/1376/files#diff-3c70a8bcbb58522dbe589e774a85bbb611adab7142bb1eacec4fb7828cb3ee06R684-R723) that just does the same error checks as what we do for persisting `channel_manager` 😄